server/src/main/java/org/elasticsearch/rest/RestUtils.java

```diff
@@ -323,4 +328,21 @@ public static TimeValue getTimeout(RestRequest restRequest) {
         assert restRequest != null;
         return restRequest.paramAsTime(REST_TIMEOUT_PARAM, null);
     }
+
+    // Remove the BWC support for the deprecated ?local parameter.
+    // NOTE: ensure each usage of this method has been deprecated for long enough to remove it.
+    @UpdateForV10(owner = UpdateForV10.Owner.DISTRIBUTED_COORDINATION)
+    public static void consumeDeprecatedLocalParameter(RestRequest request) {
+        if (request.getRestApiVersion() == RestApiVersion.V_8) {
+            return;
```

Comment:
We can't return here, we still need to consume the `local` parameter. If we don't, we get this when using the rest compat headers:

```
λ curl -vv -H "Authorization: Basic ZWxhc3RpYzpwYXNzd29yZA==" -H "Content-Type: application/vnd.elasticsearch+json;compatible-with=8" -XGET "http://localhost:9200/_component_template?local&pretty=true" -H "Accept: application/vnd.elasticsearch+json;compatible-with=8"
Note: Unnecessary use of -X or --request, GET is already inferred.
* Host localhost:9200 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:9200...
* Connected to localhost (::1) port 9200
> GET /_component_template?local&pretty=true HTTP/1.1
> Host: localhost:9200
> User-Agent: curl/8.7.1
> Authorization: Basic ZWxhc3RpYzpwYXNzd29yZA==
> Content-Type: application/vnd.elasticsearch+json;compatible-with=8
> Accept: application/vnd.elasticsearch+json;compatible-with=8
> 
* Request completely sent off
< HTTP/1.1 400 Bad Request
< X-elastic-product: Elasticsearch
< content-type: application/vnd.elasticsearch+json;compatible-with=8
< content-length: 356
< 
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "request [/_component_template] contains unrecognized parameter: [local]"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "request [/_component_template] contains unrecognized parameter: [local]"
  },
  "status" : 400
}
* Connection #0 to host localhost left intact
```

Reply:
Great catch! Thanks :)